tests/test_extract.py

```diff
@@ -51,25 +57,36 @@ def test_empty():
     # decompress empty archive
     archive = py7zr.SevenZipFile(open(os.path.join(testdata_path, 'empty.7z'), 'rb'))
     assert archive.getnames() == []
+    archive.reset()
+    _dict = archive.extractall(return_dict=True)
+    assert _dict == {}
 
 
 @pytest.mark.files
 def test_github_14(tmp_path):
     archive = py7zr.SevenZipFile(open(os.path.join(testdata_path, 'github_14.7z'), 'rb'))
     assert archive.getnames() == ['github_14']
     archive.extractall(path=tmp_path)
+    archive.reset()
+    _dict = archive.extractall(return_dict=True)
     with tmp_path.joinpath('github_14').open('rb') as f:
         assert f.read() == bytes('Hello GitHub issue #14.\n', 'ascii')
+    actual = _dict['github_14'].read()
+    assert actual == bytes('Hello GitHub issue #14.\n', 'ascii')
 
 
 @pytest.mark.files
 def _test_umlaut_archive(filename: str, target: pathlib.Path):
     archive = py7zr.SevenZipFile(open(os.path.join(testdata_path, filename), 'rb'))
     assert sorted(archive.getnames()) == ['t\xe4st.txt']
     archive.extractall(path=target)
+    archive.reset()
```

Comment:
please add  flag for helper function and new test cases
e.g.
 `def _test_umlaut_archive(filename, target, return_dict: False)`
`def test_non_solid_umlaut_dict()`